config-service-impl/src/main/java/org/hypertrace/config/service/store/ConfigDocumentKey.java

```diff
@@ -0,0 +1,27 @@
+package org.hypertrace.config.service.store;
+
+import org.hypertrace.config.service.ConfigResource;
+import org.hypertrace.core.documentstore.Key;
+
+/**
+ * Key for the Config Document
+ */
+public class ConfigDocumentKey implements Key {
+
+  private static final String SEPARATOR = ":";
+
+  private final ConfigResource configResource;
+  private final long configVersion;
```

Comment:
Should the version be part of the resource?

Reply:
I like to think of it as each resource can have multiple versions of configs associated with it. This way the APIs are also cleaner as some APIs only require the resource and not the version.